nucypher/blockchain/eth/signers.py

```diff
@@ -191,3 +216,93 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
     @validate_checksum_address
     def lock_account(self, account: str):
         return True
+
+
+class KeyStoreSigner(Signer):
+
+    IS_SIGNER_LOCAL = True
+
+    def __init__(self, uri: str):
+        super().__init__()
+
+        decoded_uri = urlparse(uri)
+        if decoded_uri.scheme not in ('key', 'keystore') or decoded_uri.netloc:
+            raise cls.InvalidSignerURI(uri)
+
+        with open(decoded_uri.path) as file_obj:
+            key = json.load(file_obj)
+
+        address = key['address']
+        if not Web3.isAddress(address):
+            raise ValueError(f"'{decoded_uri.path}' does not contain a valid address")
+
+        self.key = key
+        self._account = Web3.toChecksumAddress(address)
+        self._signer = None
+
+    def _is_my_account(self, account) -> bool:
+        return Web3.isAddress(account) and \
+            self.account == Web3.toChecksumAddress(account)
+
+    @classmethod
+    def from_signer_uri(cls, uri: str) -> 'Signer':
+        return cls(uri)
+
+    @property
+    def account(self) -> str:
+        return self._account
+
+    @validate_checksum_address
+    def is_device(self, account: str) -> bool:
+        return False
+
+    def accounts(self) -> List[str]:
+        return [self.account]
+
+    @validate_checksum_address
+    def unlock_account(self, account: str, password: str, duration: int = None) -> bytes:
+        if not self._is_my_account(account):
+            return False
+
+        if self._signer is None:
+            try:
+                self._signer = Account.from_key(
+                    Account.decrypt(self.key, password)
+                )
+            except ValueError:
+                return False
+
+            if self.account != self._signer.address:
+                raise ValueError("unexpected address '%s' recovered from private key; expected '%s'" % (
+                    self._signer.address,
+                    self.account,
+                ))
+
+        return True
+
+    @validate_checksum_address
+    def lock_account(self, account: str) -> str:
+        if not self._is_my_account(account):
+            return False
+
+        self._signer = None
+        return True
+
+    @validate_checksum_address
+    def sign_transaction(self, transaction_dict: dict) -> HexBytes:
+        # Do not include a 'to' field for contract creation.
```

Comment:
What is the expected behavior in the case of invoking this method with a locked account?

Reply:
An AttributeError will be raised as self.__signer will be None if the account is locked.

Web3Signer/sign_* methods would raise a ValueError (assuming the RPC calls return an error result).

If something cleaner is needed, I'd suggest adding another exception type to Signer (i.e. `LockedAccount`) and having all the signers throw this when signing with a locked account.

Reply:
The sign_* methods will now throw in case of locked or unknown accounts and the exception message text should be helpful in these cases.